serdes/serde-common/src/main/java/io/apicurio/registry/serde/AbstractSchemaResolver.java

```diff
@@ -229,7 +244,8 @@ private OidcAuth configureAuthWithUrl(DefaultSchemaResolverConfig config, String
             throw new IllegalArgumentException("Missing registry auth secret, set " + SerdeConfig.AUTH_CLIENT_SECRET);
         }
 
-        return new OidcAuth(tokenEndpoint, clientId, clientSecret, Optional.empty());
+        authClient = ApicurioHttpClientFactory.create(tokenEndpoint, new AuthErrorHandler());
+        return new OidcAuth(authClient, clientId, clientSecret);
```

Comment:
this is out of the scope of this task, but we could also try to share the same http client for the registry client and the oidcauth classes. And actually I think implementing that could be complex due to all the context information like the url given to the client and the error handler... but still it may be nice to do it in the future

Reply:
I think it's not that hard. We can extend the interface ApicurioHttpClient to accept a baseURL and AFAIR, both the JDK and the Vert.x client already accept that. I'll check.